test/log.ts

```diff
@@ -413,161 +435,157 @@ describe('Log', () => {
         resource: CUSTOM_RESOURCE,
       });
 
-      log.logging.loggingService.writeLogEntries = (
-        reqOpts: {},
-        gaxOpts: {}
-      ) => {
-        assert.deepStrictEqual(reqOpts, {
-          logName: log.formattedName_,
-          entries: [ENTRY],
-          resource: EXPECTED_RESOURCE,
-        });
-
-        assert.strictEqual(gaxOpts, undefined);
-      };
-
       await log.write(ENTRY, optionsWithResource);
+      assert(
+        writeLogStub.calledOnceWithExactly(
+          {
+            logName: log.formattedName_,
+            entries: [ENTRY],
+            resource: EXPECTED_RESOURCE,
+          },
+          undefined
+        )
+      );
     });
 
     it('should call gax method', async () => {
-      log.logging.loggingService.writeLogEntries = (
-        reqOpts: {},
-        gaxOpts: {}
-      ) => {
-        assert.deepStrictEqual(reqOpts, {
-          logName: log.formattedName_,
-          entries: [ENTRY],
-          resource: FAKE_RESOURCE,
-        });
-
-        assert.strictEqual(gaxOpts, undefined);
-      };
-
       await log.write(ENTRY, OPTIONS);
+      assert(
+        writeLogStub.calledOnceWithExactly(
+          {
+            logName: log.formattedName_,
+            entries: [ENTRY],
+            resource: FAKE_RESOURCE,
+          },
+          undefined
+        )
+      );
     });
 
     it('should arrify & decorate the entries', async () => {
       const decoratedEntries = [] as Entry[];
-
-      log.decorateEntries_ = (entries: Entry[]) => {
-        assert.strictEqual(entries[0], ENTRY);
-        return decoratedEntries;
-      };
-
-      // tslint:disable-next-line no-any
-      log.logging.loggingService.writeLogEntries = (
-        // tslint:disable-next-line no-any
-        reqOpts: any,
-        gaxOpts: {}
-      ) => {
-        assert.strictEqual(reqOpts.entries, decoratedEntries);
-      };
+      decorateEntriesStub.returns(decoratedEntries);
+      const arrifiedEntries = [ENTRY] as Entry[];
 
       await log.write(ENTRY, OPTIONS);
+      assert(decorateEntriesStub.calledOnceWithExactly(arrifiedEntries));
+      assert(
+        writeLogStub.calledOnceWith(
+          sinon.match({
+            entries: arrifiedEntries,
```

Comment:
I think this should check that what we returned from `decorateEntries_` was used in the API request.

Reply:
The original had `decorateEntries_` stubbed out, but I can still do this if you think it's necessary.

Reply:
Yeah, we ultimately want to know (whether in this test or a separate one) that the decorated entries were used.

Reply:
good point. performed this in the "should truncate the entries after decorating" test, and split into 3 separate tests.